ctest-next/src/generation/generator.rs

```diff
@@ -0,0 +1,20 @@
+use std::{error::Error, path::Path};
+
+use crate::expand;
+
+pub struct TestGenerator;
+
+impl TestGenerator {
+    pub fn new() -> Self {
+        Self
+    }
+
+    pub fn generate<P: AsRef<Path>>(
+        &self,
+        crate_path: P,
+        _output_file_path: P,
+    ) -> Result<(), Box<dyn Error>> {
```

Comment:
Could you add

```rust
type Error = Box<dyn std::error::Error>;
type Result<T, E = Error> = std::result::Result<T, E>;
```

And then import those? Rather than retyping the definitions everywhere.